### PR TITLE
[bitnami/memcached] Fix ClusterIP invalid value and /tmp volumeMount using high-availability 

### PR DESCRIPTION
**Pull request by SergeSpinoza**
**Description of the change**

This changes fix 2 errors: 

1. When you try to deploy memcached via this helm chart and use `architecture: high-availability` and `service.type: LoadBalancer` - you have error: 

```
Release "memcached" does not exist. Installing it now.
Error: Service "memcached" is invalid: spec.clusterIP: Invalid value: "None": may not be set to 'None' for LoadBalancer services
```
I fix it via check type of service. 

2. When you try to deploy memcached via this helm chart and use `architecture: high-availability` and `persistence.enabled: false` - you have error with memcached - it can't starts, in memcached log you see: 

```
memcached 12:14:56.03
memcached 12:14:56.03 Welcome to the Bitnami memcached container
memcached 12:14:56.03 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-memcached
memcached 12:14:56.03 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-memcached/issues
memcached 12:14:56.04

/run.sh: line 28: cannot create temp file for here-document: Read-only file system
```
My second commit is fix this. 

**Benefits**

Fix 2 errors. 

**Possible drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/memcached/templates/service.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ metadata:
   {{- end }}
 spec:
   type: {{ .Values.service.type }}
-  {{- if eq .Values.architecture "high-availability" }}
+  {{- if and (eq .Values.architecture "high-availability") (ne .Values.service.type "LoadBalancer") }}
```
Comment (FraPazGal) [inline bitnami/memcached/templates/service.yaml:11]: We should also take into account the use case when `service.type=NodePort`. We could change the condition to something like:

```suggestion
  {{- if and (eq .Values.architecture "high-availability") (eq .Values.service.type "ClusterIP") }}
```

**Review by FraPazGal — COMMENTED** [review]
Hi @SergeSpinoza,

I'm having trouble to reproduce the `/tmp` error. Doing a `helm install memcached bitnami/memcached --set a--set architecture=high-availability ` doesn't seem to throw any errors. Did you change any other parameter?

Also, I added a comment to your changes, could you please give it a look?

**Comment by SergeSpinoza** [comment]
> Hi @SergeSpinoza,
> 
> I'm having trouble to reproduce the `/tmp` error. Doing a `helm install memcached bitnami/memcached --set a--set architecture=high-availability ` doesn't seem to throw any errors. Did you change any other parameter?
> 
> Also, I added a comment to your changes, could you please give it a look?

Hi! I also change `persistence.enabled` to `false`

File: bitnami/memcached/templates/service.yaml (hunk 1 above)
Comment (SergeSpinoza) [reply bitnami/memcached/templates/service.yaml:11]: Yes, it will probably be more correct.

**Comment by FraPazGal** [comment]
Yes, `persistence.enabled` is set to `false` by default in `values.yaml`, but I can't reproduce the error still. Did you change the `securityContext` parameter?

**Comment by SergeSpinoza** [comment]
> Yes, `persistence.enabled` is set to `false` by default in `values.yaml`, but I can't reproduce the error still. Did you change the `securityContext` parameter?

No, I don't change it. 

This is my values.yaml with changes:

```
## Global Docker image parameters
## Please, note that this will override the image parameters, including dependencies, configured to use the global value
## Current available global Docker image parameters: imageRegistry and imagePullSecrets
##
# global:
#   imageRegistry: myRegistryName
#   imagePullSecrets:
#     - myRegistryKeySecretName
#   storageClass: myStorageClass

## Bitnami Memcached image version
## ref: https://hub.docker.com/r/bitnami/memcached/tags/
##
image:
  registry: docker.io
  repository: bitnami/memcached
  tag: 1.6.9-debian-10-r27
  ## Specify a imagePullPolicy
  ## Defaults to 'Always' if image tag is 'latest', else set to 'IfNotPresent'
  ## ref: http://kubernetes.io/docs/user-guide/images/#pre-pulling-images
  ##
  pullPolicy: IfNotPresent
  ## Optionally specify an array of imagePullSecrets.
  ## Secrets must be manually created in the namespace.
  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
  ##
  # pullSecrets:
  #   - myRegistryKeySecretName

  ## Set to true if you would like to see extra information on logs
  ##
  debug: false

## Extra environment vars to pass.
## ref: https://github.com/bitnami/bitnami-docker-memcached#configuration
extraEnv: []

## String to partially override common.names.fullname template (will maintain the release name)
##
# nameOverride:

## Number of containers to run
replicaCount: 2

## String to fully override common.names.fullname template
##
# fullnameOverride:

# Cluster domain
clusterDomain: cluster.local

## Memcached architecture. Allowed values: standalone or high-availability
##
architecture: high-availability

## Memcached admin user
## ref: https://github.com/bitnami/bitnami-docker-memcached#creating-the-memcached-admin-user
##
# memcachedUsername:

## Memcached admin password
## ref: https://github.com/bitnami/bitnami-docker-memcached#creating-the-memcached-admin-user
##
# memcachedPassword:

## Service parameters
##
##
service:
  ## Service type
  ##
  # type: ClusterIP
  type: LoadBalancer
  ## Memcached port
  ##
  port: 11211
  ## Specify the nodePort value for the LoadBalancer and NodePort service types.
  ## ref: https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport
  ##
  nodePort: ""
  ## Set the LoadBalancer service type to internal only.
  ## ref: https://kubernetes.io/docs/concepts/services-networking/service/#internal-load-balancer
  ##
  loadBalancerIP: 192.168.136.212
  ## Annotations for the Memcached service
  ##
  annotations: {}

## Memcached containers' resource requests and limits
## ref: http://kubernetes.io/docs/user-guide/compute-resources/
##
resources:
  # We usually recommend not to specify default resources and to leave this as a conscious
  # choice for the user. This also increases chances charts run on environments with little
  # resources, such as Minikube. If you do want to specify resources, uncomment the following
  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
  limits:
    memory: 3096Mi
    cpu: 500m
  requests:
    memory: 3096Mi
    cpu: 300m

## Pod Security Context
## ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
##
securityContext:
  enabled: true
  fsGroup: 1001
  runAsUser: 1001
  readOnlyRootFilesystem: true

## Pod annotations
## ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
##
podAnnotations: {}

## Pod affinity preset
## ref: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity
## Allowed values: soft, hard
##
podAffinityPreset: ""

## Pod anti-affinity preset
## Ref: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity
## Allowed values: soft, hard
##
podAntiAffinityPreset: hard

## Node affinity preset
## Ref: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity
## Allowed values: soft, hard
##
nodeAffinityPreset:
  ## Node affinity type
  ## Allowed values: soft, hard
  type: ""
  ## Node label key to match
  ## E.g.
  ## key: "kubernetes.io/e2e-az-name"
  ##
  key: ""
  ## Node label values to match
  ## E.g.
  ## values:
  ##   - e2e-az1
  ##   - e2e-az2
  ##
  values: []

## Affinity for pod assignment. Evaluated as a template.
## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
## Note: podAffinityPreset, podAntiAffinityPreset, and nodeAffinityPreset will be ignored when it's set
##
affinity: {}

## Node labels for pod assignment. Evaluated as a template.
## ref: https://kubernetes.io/docs/user-guide/node-selection/
##
nodeSelector: {}

## Tolerations for pod assignment. Evaluated as a template.
## ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
##
tolerations: []

## Pod priority
## ref: https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/
##
# priorityClassName: ""

## Persistence - used for dumping and restoring states between recreations
## Ref: https://github.com/memcached/memcached/wiki/WarmRestart
persistence:
  enabled: false
  ## Persistent Volume Storage Class
  ## If defined, storageClassName: <storageClass>
  ## If set to "-", storageClassName: "", which disables dynamic provisioning
  ## If undefined (the default) or set to null, no storageClassName spec is
  ##  set, choosing the default provisioner.  (gp2 on AWS, standard on
  ##  GKE, AWS & OpenStack)
  ##
  # storageClass: "-"
  ## Persistent Volume Claim annotations
  ##
  annotations: {}
  ## Persistent Volume Access Mode
  ##
  accessModes:
    - ReadWriteOnce
  ## Persistent Volume size
  ##
  size: 8Gi


## Args for running memcached
## Ref: https://github.com/memcached/memcached/wiki/ConfiguringServer
arguments:
  - /run.sh
  - -c 20480
  - -m 2560
  # - -m <maxMemoryLimit>
  # - -I <maxItemSize>
  # - -vv

## Prometheus Exporter / Metrics
##
metrics:
  enabled: false
  ## Bitnami Memcached Prometheus Exporter image
  ## ref: https://hub.docker.com/r/bitnami/memcached-exporter/tags/
  ##
  image:
    registry: docker.io
    repository: bitnami/memcached-exporter
    tag: 0.8.0-debian-10-r15
    pullPolicy: IfNotPresent
    ## Optionally specify an array of imagePullSecrets.
    ## Secrets must be manually created in the namespace.
    ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
    ##
    # pullSecrets:
    #   - myRegistryKeySecretName
  ## Metrics exporter pod Annotation and Labels
  ## ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
  ##
  podAnnotations:
    prometheus.io/scrape: "true"
    prometheus.io/port: "9150"
  ## Memcached Prometheus exporter resource requests and limits
  ## ref: http://kubernetes.io/docs/user-guide/compute-resources/
  ##
  resources:
    # We usually recommend not to specify default resources and to leave this as a conscious
    # choice for the user. This also increases chances charts run on environments with little
    # resources, such as Minikube. If you do want to specify resources, uncomment the following
    # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
    limits: {}
    #   cpu: 100m
    #   memory: 128Mi
    requests: {}
    #   cpu: 100m
    #   memory: 128Mi
  service:
    ## Service type
    ##
    type: ClusterIP
    ## Memcached Prometheus exporter port
    ##
    port: 9150
    ## Annotations for the Prometheus metrics service
    ##
    annotations:
      prometheus.io/scrape: "true"
      prometheus.io/port: "{{ .Values.metrics.service.port }}"
  ## Prometheus Operator ServiceMonitor configuration
  ##
  serviceMonitor:
    enabled: false
    ## Namespace in which Prometheus is running
    ##
    # namespace: monitoring

    ## Interval at which metrics should be scraped.
    ## ref: https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#endpoint
    ##
    # interval: 10s

    ## Timeout after which the scrape is ended
    ## ref: https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#endpoint
    ##
    # scrapeTimeout: 10s

    ## ServiceMonitor selector labels
    ## ref: https://github.com/bitnami/charts/tree/master/bitnami/prometheus-operator#prometheus-configuration
    ##
    # selector:
    #   prometheus: my-prometheus
```

**Comment by FraPazGal** [comment]
Oh, I found the missing parameter, you were setting `readOnlyRootFilesystem=true`. Indeed, I was able to reproduce the issue and it seems it isn't behaving a it does when `architecture=standalone`.

I'll merge this once the change what we discussed yesterday on `service.yaml` is committed. Thanks for the PR, really appreciated!

**Review by FraPazGal — APPROVED** [review]
LGTM!

**Pull request merged**